src/libstd/sys/unix/thread.rs

```diff
@@ -71,7 +71,7 @@ impl Thread {
         };
 
         extern fn thread_start(main: *mut libc::c_void) -> *mut libc::c_void {
-            unsafe { start_thread(main); }
+            unsafe { start_thread(main, pthread_self() as u32); }
```

Comment:
On Linux pthread_t is `unsigned long`, thus u64 on amd64. There is no portable way to get a numeric ID from pthreads (`pthread_getthreadid_np` is not portable).

I'm starting to think that using the address of the Inner struct is way simpler and would totally work.